openedx/core/djangoapps/content/block_structure/tests/test_signals.py

```diff
@@ -41,6 +46,17 @@ def test_course_update(self):
             updated_block_structure.get_xblock_field(self.course_usage_key, 'display_name')
         )
 
+    @ddt.data(True, False)
+    @patch('openedx.core.lib.block_structure.manager.BlockStructureManager.clear')
+    def test_cache_invalidation(self, invalidate_cache_enabled, mock_bs_manager_clear):
+        test_display_name = "Jedi 101"
+
+        with override_switch(INVALIDATE_CACHE_ON_PUBLISH_SWITCH, active=invalidate_cache_enabled):
+            self.course.display_name = test_display_name
+            self.store.update_item(self.course, self.user.id)
+
+        self.assertEquals(mock_bs_manager_clear.called, invalidate_cache_enabled)
+
```

Comment:
Yes.  This test is `ddt`ed so both cases are tested according to the value passed in for `invalidate_cache_enabled`.